test/odbc/psqlodbc/test_pg_varchar.cpp

```diff
@@ -0,0 +1,655 @@
+
+  const vector<int> LENGTH_EXPECTED_8000 = {4, 8000};
+  const vector<int> PRECISION_EXPECTED_8000 = {0, 0};
+  const vector<int> SCALE_EXPECTED_8000 = {0, 0};
+  const vector<string> NAME_EXPECTED_8000 = {"int4", "unknown"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_8000);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_8000.size(), COL1_NAME, LENGTH_EXPECTED_8000, 
+    PRECISION_EXPECTED_8000, SCALE_EXPECTED_8000, NAME_EXPECTED_8000);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<int> LENGTH_EXPECTED_20 = {4, 20};
+  const vector<int> PRECISION_EXPECTED_20 = {0, 0};
+  const vector<int> SCALE_EXPECTED_20 = {0, 0};
+  const vector<string> NAME_EXPECTED_20 = {"int4", "unknown"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20.size(), COL1_NAME, LENGTH_EXPECTED_20, 
+    PRECISION_EXPECTED_20, SCALE_EXPECTED_20, NAME_EXPECTED_20);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Varchar, Table_Create_Fail) {
+  const vector<vector<pair<string, string>>> invalid_columns {
+    {{"invalid1", DATATYPE + "(-1)"}},
+    {{"invalid1", DATATYPE + "(0)"}},
+    {{"invalid1", DATATYPE + "(NULL)"}}
+  };
+  testTableCreationFailure(ServerType::PSQL, TABLE_NAME, invalid_columns);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Varchar, Insertion_Success) {
+  const vector<string> inserted_values = {
+    "NULL", 
+    STRING_1,
+    "" 
+  };
+
+  const vector<string> expected = {
+    "NULL", 
+    STRING_1,
+    "" 
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_1);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values, expected);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> inserted_values_8000 = {
+    "NULL", // NULL values
+    "",
+    STRING_1,
+    STRING_8000,
+    STRING_20
+  };
+
+  const vector<string> expected_8000 = {
+    "NULL", // NULL values
+    "",
+    STRING_1,
+    STRING_8000,
+    STRING_20
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_8000);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_8000, expected_8000);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+
+  const vector<string> inserted_values_20 = {
+    "NULL", // NULL values
+    "",
+    STRING_1,
+    STRING_20
+  };
+
+  const vector<string> expected_20 = {
+    "NULL", // NULL values
+    "",
+    STRING_1,
+    STRING_20
+  };
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS_20);
+  testInsertionSuccess(ServerType::PSQL, TABLE_NAME, COL1_NAME, inserted_values_20, expected_20);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+ 
+}
+
+TEST_F(PSQL_DataTypes_Varchar, Insertion_Failure) {
+  const vector<string> inserted_values = {
```

Comment:
nit: as a general rule, please ensure that all characters in `const` variables are capitalized.